modules/javafx.graphics/src/main/java/com/sun/javafx/iio/common/ImageDescriptor.java

```diff
@@ -38,12 +36,9 @@ public class ImageDescriptor implements ImageFormatDescription {
 
     public ImageDescriptor(String formatName, String[] extensions, Signature[] signatures, String[] mimeSubtypes) {
         this.formatName = formatName;
-        this.extensions = Collections.unmodifiableList(
-                                          Arrays.asList(extensions));
-        this.signatures = Collections.unmodifiableList(
-                                          Arrays.asList(signatures));
-        this.mimeSubtypes = Collections.unmodifiableList(
-                                          Arrays.asList(mimeSubtypes));
+        this.extensions = List.of(extensions);
+        this.signatures = List.of(signatures);
+        this.mimeSubtypes = List.of(mimeSubtypes);
```

Comment:
While this is not equivalent since changing the backing array will not change the resulting list anymore, I would consider the old code a bug and this is correct.

Note that in `FileChooser` care is taken to create a copy of the supplied array before using it to create a list.

Additionally, care must be taken that all the callers don't have `null` elements in the arrays they pass. I checked it and it's fine (and should probably be disallowed, which is good now).

By the way, this class should be a `record`, and all its subtypes, which are not really subtypes but just configured instances, should be modified accordingly. This is out of scope though.